### PR TITLE
[Tiered Caching] Fixing flaky tiered cache test

### PR DESCRIPTION
**Pull request by sgup432**
### Description
Fixing flaky tiered cache test. The issue was basically a typo where we were incorrectly passing weigher(keyValueSize) to underlying onHeap cache.

- Also fixed other issue where we only call removal listener when a item is removed from disk cache as well.

### Related Issues
Resolves https://github.com/opensearch-project/OpenSearch/issues/12593

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- ~[x] New functionality has been documented.~
  - ~[x] New functionality has javadoc added~
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
- ~[ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))~
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by sgup432** [comment]
@reta @msfroh Can you take a look at this fix?
Also let me know if changelog is needed for this?

**Review by msfroh — APPROVED** [review]
Oh, of course.... removing from the OnHeapCache doesn't mean removing from the tiered cache, because it just spills over to the disk cache.

File: modules/cache-common/src/test/java/org/opensearch/cache/common/tier/TieredSpilloverCacheTests.java [hunk 1]
```diff
@@ -291,7 +287,7 @@ public void testComputeIfAbsentWithEvictionsFromOnHeapCache() throws Exception {
             tieredSpilloverCache.computeIfAbsent(key, tieredCacheLoader);
         }
         long actualDiskCacheSize = tieredSpilloverCache.getDiskCache().count();
-        assertEquals(actualDiskCacheSize, removalListener.evictionsMetric.count()); // Evictions from onHeap equal to
+        // assertEquals(actualDiskCacheSize, removalListener.evictionsMetric.count()); // Evictions from onHeap equal to
```
Comment (kkmr) [inline modules/cache-common/src/test/java/org/opensearch/cache/common/tier/TieredSpilloverCacheTests.java:290]: remove unused variable?

**Review by kkmr — APPROVED** [review]

File: modules/cache-common/src/test/java/org/opensearch/cache/common/tier/TieredSpilloverCacheTests.java (hunk 1 above)
Comment (sgup432) [reply modules/cache-common/src/test/java/org/opensearch/cache/common/tier/TieredSpilloverCacheTests.java]: Sure, removed!

**Pull request merged**